kernel/kernel-api/src/main/java/io/delta/kernel/internal/actions/Protocol.java

```diff
@@ -105,23 +141,261 @@ public Row toRow() {
     Map<Integer, Object> protocolMap = new HashMap<>();
     protocolMap.put(0, minReaderVersion);
     protocolMap.put(1, minWriterVersion);
-    protocolMap.put(2, stringArrayValue(readerFeatures));
-    protocolMap.put(3, stringArrayValue(writerFeatures));
+    protocolMap.put(2, stringArrayValue(new ArrayList<>(readerFeatures)));
+    protocolMap.put(3, stringArrayValue(new ArrayList<>(writerFeatures)));
 
     return new GenericRow(Protocol.FULL_SCHEMA, protocolMap);
   }
 
+  /////////////////////////////////////////////////////////////////////////////////////////////////
+  /// Public methods related to table features interaction with the protocol                    ///
+  /////////////////////////////////////////////////////////////////////////////////////////////////
+  /**
+   * Get the set of features that are implicitly enabled by the protocol. Features are implicitly
+   * enabled if they reader and/or writer version is less than the versions that support the
+   * explicit features specifying in `readerFeatures` and `writerFeatures` sets.
```

Comment:
nit: `specified` instead of `specifying`

Can you also give an example? IMO a succinct clear example is worth 1000 words of description. Ditto for other methods -- short examples would make it super clear.